test-project/project.clj

```diff
@@ -6,10 +6,12 @@
   :middleware [lein-git-down.plugin/inject-properties]
   :git-down {cheshire {:coordinates dakrone/cheshire}
              demo-deps {:coordinates puredanger/demo-deps}
-             com.cemerick/pomegranate {:coordinates cemerick/pomegranate}}
+             com.cemerick/pomegranate {:coordinates cemerick/pomegranate}
+             cljfmt {:coordinates weavejester/cljfmt :manifest-root "cljfmt"}}
   :dependencies [[com.cemerick/pomegranate "a8d0ef79d6cbbd9392dbe7f824e32dc60d46e0c0"] ;; pom based
                  [cheshire "c79ebaa3f56c365a1810f80617c80a3b62999701"]                 ;; lein based
                  [demo-deps "19d387dc11d804ab955207a263dfba5dbd15bf2c"]                ;; deps based
                  [clj-time "66ea91e68583e7ee246d375859414b9a9b7aba57"]                 ;; multiple
+                 [cljfmt "806e43b7a7d4e22b831d796f107f135d8efc986a"]                   ;; contains hooks
```

Comment:
@manderson202 :+1: for including an example that has hooks and for all the detailed comments explaining the out-of-band behavior.

Are there particular before (failing) and after (fixed) states observable by running the test suite? I left this dependency in place and then commented out a couple different parts of this fix (including the `locking` call) and in all scenarios the test suite passes.

Reply:
Hm. Interesting. I could add in the dependency from the issue that explicitly caused it to fail. That was always reproducible (as long as starting from clean .m2/.gitlibs) and is what I used to verify my fix.

Reply:
FYI: I just tried the before fix with only adding cljfmt and it failed as expected. Let me know if you want to discuss.

Reply:
Added an additional check for the test-project.jar existing. This would fail if the fix was not in place.